### PR TITLE
fix(trace_stats): ensures stats respect the report_hostname configuration

### PR DESCRIPTION
**Pull request by mabdinur**
This fix will mitigate cardinality issues in internal services. The hostname will be set by the Datadog Agent when `DD_TRACE_REPORT_HOSTNAME=False`.

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed. If no release note is required, add label `changelog/no-changelog`.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)
- [x] If this PR touches code that signs or publishes builds or packages, or handles credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.
- [x] This PR doesn't touch any of that.

### CODE REVIEW

File: tests/integration/test_trace_stats.py [hunk 1]
```diff
@@ -97,13 +98,35 @@ def test_compute_stats_default_and_configure(run_python_code_in_subprocess, envv
...
+
+assert stats_processor is not None
+assert stats_processor._hostname == "" # report_hostname is disabled by default
 """,
         env=env,
     )
     assert status == 0, out + err
 
 
+@mock.patch("ddtrace.internal.hostname.get_hostname")
```
Comment (mabdinur) [inline tests/integration/test_trace_stats.py:115]:
```suggestion
@mock.patch("ddtrace.internal.processor.stats.get_hostname")
```

**Comment by mabdinur** [comment]
We will need to update system-tests: https://github.com/DataDog/dd-trace-py/actions/runs/7559749497/job/20584255415?pr=8130

**Comment by mabdinur** [comment]
This should fix the failing system tests: https://github.com/DataDog/system-tests/pull/2042

**Review by majorgreys — APPROVED** [review]

**Comment by majorgreys** [comment]
@mabdinur add backports

**Pull request merged**